CloudComputingCourse/Spirit
# Adding support to make comments private until a certain date, specifi…

…ed per category

Notes:
1. Introduces a comments_visible_after date for each Category, default to None
2. Comments are hidden until the specified date
3. Before the specified date, comments are only visible to the original poster, admins, and moderators
4. Additionally, replies can not be posted by other users until after the visible after date
5. Adds Bootstrap 3.7 for Alert message support

Testing:
1. Make post with admin account
2. Make post with non-admin account
3. View other user's post with non-admin account; only see `Public after date` message and can not reply
4. View other user's post with admin account; see `Public after date` message, comments are visible, and can reply
5. View own user's post; see `Public after date` message, comments are visible, and can reply
